src/gsi/cplr_get_fv3_regional_ensperts.f90

```diff
@@ -460,6 +574,155 @@ subroutine general_read_fv3_regional(this,fv3_filenameginput,g_ps,g_u,g_v,g_tv,g
 
   return       
   end subroutine general_read_fv3_regional
+
+  subroutine general_read_fv3_regional_dirZDA(this,fv3_filenameginput,g_ql,g_qi,g_qr,g_qs,g_qg,g_qnr,g_w)
+  !$$$  subprogram documentation block
+  !     firstly copied from general_read_arw_regional           .      .    .                                       .
+  ! subprogram:    general_read_fv3_regional_dirZda  read fv3sar model ensemble members
+  !   prgmmr: Ting             org: emc/ncep            date: 2018
+  !
+  ! abstract: read ensemble members from the fv3 model netcdf format (dynamic and tracer)
+  !           for use with hybrid ensemble option.
+  !
+  ! program history log:
+  !   2018-  Ting      - intial versions  
+  !   2019-03-13  CAPS(C. Tong)   - Port direct radar DA capabilities
+  !   2021-05-05  CAPS(J. Park)   - Modified to read hydrometeors and w only
+  !                                 
+  !   input argument list:
+  !
+  !   output argument list:
+  !
+  ! attributes:
+  !   language: f90
+  !   machine:  ibm RS/6000 SP
+  !
+  !$$$ end documentation block
+  
+      use netcdf, only: nf90_nowrite
+      use netcdf, only: nf90_open,nf90_close
+      use netcdf, only: nf90_inq_dimid,nf90_inquire_dimension
+      use netcdf, only: nf90_inq_varid,nf90_inquire_variable,nf90_get_var
+      use kinds, only: r_kind,r_single,i_kind
+      use gridmod, only: eta1_ll,eta2_ll
+      use constants, only: zero,one,fv,zero_single,one_tenth,h300
+      use hybrid_ensemble_parameters, only: grd_ens,q_hyb_ens
+      use hybrid_ensemble_parameters, only: fv3sar_ensemble_opt 
+
+      use mpimod, only: mpi_comm_world,mpi_rtype,mype
+      use netcdf_mod, only: nc_check
+      use gsi_rfv3io_mod,only: type_fv3regfilenameg
+      use gsi_rfv3io_mod,only:n2d 
+      use gsi_rfv3io_mod,only:mype_qr,mype_qs,mype_qg,mype_qnr,mype_w
+      use gsi_rfv3io_mod,only:mype_ql,mype_qi
+      use constants, only: half,zero
+      use gsi_rfv3io_mod, only: gsi_fv3ncdf_read 
+      use gsi_rfv3io_mod, only: gsi_fv3ncdf_read_v1
+      use directDA_radaruse_mod, only: l_use_cvpqx, cvpqx_pval, cld_nt_updt
+      use directDA_radaruse_mod, only: l_cvpnr, cvpnr_pval
+  
+      implicit none
+  !
+  ! Declare passed variables
+      class(get_fv3_regional_ensperts_class), intent(inout) :: this
+      type (type_fv3regfilenameg)                  , intent (in)   :: fv3_filenameginput
+      real(r_kind),dimension(grd_ens%lat2,grd_ens%lon2,grd_ens%nsig),intent(out)::g_ql,g_qi,g_qr
+      real(r_kind),dimension(grd_ens%lat2,grd_ens%lon2,grd_ens%nsig),intent(out)::g_qs,g_qg,g_qnr,g_w
+  !
+  ! Declare local parameters
+      real(r_kind),parameter:: r0_01 = 0.01_r_kind
+      real(r_kind),parameter:: r10   = 10.0_r_kind
+      real(r_kind),parameter:: r100  = 100.0_r_kind
+  !
+  !   Declare local variables
+      
+      integer(i_kind):: i,j,k,kp
+      integer(i_kind) iderivative
+  
+      
+      logical ice
```

Comment:
The variable, ice, has been identified as unused.  Please remove unused variable.

Reply:
We removed that variable as you suggested.
https://github.com/NOAA-EMC/GSI/pull/142/files#diff-86aea35b64364db7913dd1eb935d98f572e0601c925defcb5595748644029235R642